index.js

```diff
@@ -57,12 +57,16 @@ function resolveUrlLoader(content, sourceMap) {
   loader.cacheable();
 
   // incoming source-map
-  var sourceMapConsumer, contentWithMap;
+  var sourceMapConsumer, contentWithMap, sourceRoot;
   if (sourceMap) {
 
     // sass-loader outputs source-map sources relative to output directory so start our search there
     try {
       relativeToAbsolute(sourceMap.sources, outputPath, resolvedRoot);
+      // There are now absolute paths in the source map so we don't need it anymore
+      // However, later when we go back to relative paths, we need to add it again
+      sourceRoot = sourceMap.sourceRoot;
+      sourceMap.sourceRoot = null;
     } catch (exception) {
```

Comment:
Given the number of variants of `sass-loader` that people are using I really want the `resolve-url-loader` to automatically adjust to the source root scheme.

I was thinking a double `try`-`catch` block, one for the new source-root scheme and the second for the old one.

Do you think that it would work?